core/domain/platform_feature_services.py

```diff
@@ -140,7 +153,7 @@ def update_feature_flag_rules(
 # dev or prod. There should be another mode 'test' added for QA testing,
 # once it is added, this function needs to be updated to take that into
 # consideration.
-def _get_server_mode():
+def _get_server_mode() -> platform_parameter_domain.ServerModes:
```

Comment:
Shouldn't the return type be ServerMode (does the name need changing -- is it one mode or multiple modes being returned)?

Reply:
Here 2 server modes are returned ( DEV and production ) based on the condition. 

Also, `ServerModes` is defined as an Enum class. 
To align the name here with the return value we have to change the name of `ServerModes` to `ServerMode` everywhere, which may result in a mismatch in other places.

Reply:
What's our condition for other enums in the codebase? Let's standardize.

Reply:
I found that https://github.com/oppia/oppia/blob/1bd5ab86be391bb9ee896cfe0514ab3ab8093665/core/domain/platform_parameter_domain.py#L164
here also `ServerMode` sounds more reasonable. So, I changed servermodes in all places.

Reply:
can you please elaborate a bit more what is your means by -- `condition for other enums in the codebase`

Reply:
Ah, I just meant checking other enums to see what is done there.

But your changes look good to me.